### PR TITLE
Allow users to block XML-RPC completely (kind of) (Closes: #456)

### PR DESCRIPTION
**Pull request by ottok**
This still leaves some methods available.

Before:

```
curl -k -X POST https://wordpress.local/xmlrpc.php -d '<?xml version="1.0" encoding="utf-8"?><methodCall><methodName>system.listMethods</methodName><params></params></methodCall>'
<?xml version="1.0" encoding="UTF-8"?>
<methodResponse>
  <params>
    <param>
      <value>
      <array><data>
  <value><string>system.multicall</string></value>
  <value><string>system.listMethods</string></value>
  <value><string>system.getCapabilities</string></value>
  <value><string>demo.addTwoNumbers</string></value>
  <value><string>demo.sayHello</string></value>
  <value><string>pingback.extensions.getPingbacks</string></value>
  <value><string>pingback.ping</string></value>
  <value><string>mt.publishPost</string></value>
...
(list of tens of methods)
```

After:

```
curl -k -X POST https://wordpress.local/xmlrpc.php -d '<?xml version="1.0" encoding="utf-8"?><methodCall><methodName>system.listMethods</methodName><params></params></methodCall>'
<?xml version="1.0" encoding="UTF-8"?>
<methodResponse>
  <params>
    <param>
      <value>
      <array><data>
  <value><string>system.multicall</string></value>
  <value><string>system.listMethods</string></value>
  <value><string>system.getCapabilities</string></value>
</data></array>
      </value>
    </param>
  </params>
</methodResponse>
```

Also fix deprecation notices such as:

  PHP Deprecated:  Non-static method Seravo\Security_Restrictions::maybe_block_xml_rpc()
  should not be called statically in /data/wordpress/htdocs/wordpress/wp-includes/class-wp-hook.php
  on line 287

### CODE REVIEW

**Comment by ottok** [comment]
Before:
image

After:
image

Request for comments @ypcs and @elguitar

**Comment by ottok** [comment]
Rolling this out would increase the priority of #472

**Comment by ottok** [comment]
TODO: Default to run `add_filter('xmlrpc_enabled', array( __CLASS__, 'maybe_block_xml_rpc' ));` on all sites that don't have it explicitly enabled.

**Comment by ottok** [comment]
This should be ready for merge now. OK?

File: languages/seravo-fi.po [hunk 1]
```diff
@@ -1016,26 +1011,34 @@ msgid "Brute-force login protection"
...
+"Estää kirjautumisyrityksen XML-RPC:n kautta. Ei vaikuta Jetpack-lisäosan "
+"toimintaan, koska sen IP-osoitteet ovat erikseen sallittu."
 
 #: modules/security.php:228
+msgid "Completely disable XML-RPC"
+msgstr "Estä XML-RPC toiminta kokonaan"
+
+#: modules/security.php:229
+msgid "This setting prevents XML-RPC from responding to any methods at all."
+msgstr "Tämä asetus tekee, että XML-RPC ei vastaa mihinkään metodikutsuun."
```
Comment (ypcs) [inline languages/seravo-fi.po:1031]: > Tämä asetus estää XML-RPC:tä vastaamasta kutsuihin

tms.? "Tämä asetus tekee" on musta vähän huonoa kieltä.

File: languages/seravo-fi.po [hunk 2]
```diff
@@ -1693,6 +1696,7 @@ msgstr "Testaa nykyisen sivun nopeus"
 msgid ""
 "This file couldn't be sanitized so for security reasons it wasn't uploaded"
 msgstr ""
+"Tiedostonimeä ei voitu putsata, joten tietoturvasyistä sitä ei tallennettu"
```
Comment (ypcs) [inline languages/seravo-fi.po:1699]: putsata > siistiä?

File: lib/security-restrictions.php [hunk 3]
```diff
@@ -69,6 +74,10 @@ function ( $public_query_vars ) {
 
     }
 
+    public static function remove_xmlrpc_methods() {
```
Comment (ypcs) [inline lib/security-restrictions.php:77]: Maybe mention https://developer.wordpress.org/reference/hooks/xmlrpc_methods/  here, easier to see what this does / should return etc.

**Review by ypcs — COMMENTED** [review]
Option uses now much better naming (= matches how other options have been named). Some minor notes added as inline comments.

(Didn't test this feature.)

**Comment by ottok** [comment]
Comments addressed, PR updated and merged. Thanks @ypcs !

**Pull request merged**